xmodule/video_block/video_block.py

```diff
@@ -645,6 +645,9 @@ def editor_saved(self, user, old_metadata, old_content):  # lint-amnesty, pylint
                 if val_youtube_id and self.youtube_id_1_0 != val_youtube_id:
                     self.youtube_id_1_0 = val_youtube_id
 
+            if not self.edx_video_id and self.youtube_id_1_0:
+                self.thumbnail = f"https://img.youtube.com/vi/{self.youtube_id_1_0}/sddefault.jpg"
```

Comment:
The only probably I can see with this is with the edge case in which:

1. The user explicitly sets the thumbnail to a value in OLX; and...
2. The user changes some video metadata, but not the actual YouTube video location.

In that case, we would not want to override the explicit thumbnail they set with the YouTube default.

Whatever you end up writing though, please add thorough comments explaining why this is okay to do, since this logic would not be obvious to someone who isn't familiar with some of the weirdness of `VideoBlock`.

Reply:
Then we can also add `and not self.thumbnail` to the check?

Reply:
> Then we can also add `and not self.thumbnail` to the check?

This will prevent the thumbnail from being updated if the YouTube URL changes. Also, we are not cleaning the thumbnail if we change the URL (i.e. from a YouTube video for a stream)

Reply:
Ah, right. Then I think it's better to never _save_ this thumbnail value into the `thumbnail` field - besides that means that existing videos without thumbnails won't show up unless they are edited and saved. Instead, every time `index_dictionary()` is called, return `self.thumbnail or (autodetect youtube URL if applicable) or None` but don't save it into the `self.thumbnail` field.

Reply:
The advantage of saving it is that it can be used in other places, like the block editor (which works out of the box).
![image](https://github.com/user-attachments/assets/f1abeb98-14c3-483b-8a2c-b7223a566d77)

Another option would be to create a new thumbnail_youtube field to store this data and add a getter to the thumbnail property using the thumbnail field as a fallback.

```python
def get_thumbnail(self):
  return self.thumbnail or self.thumbnail_youtube
```

Reply:
_If_ we store it and the user change the YouTube URL, will we be able to automatically update the `thumbnail_youtube` field?

Reply:
Yes. We already have the logic to handle it with the `youtube_id_1_0` field.